### DISCUSSION TITLE
Aiden Bai - Million.js

### DISCUSSION THREAD

**Discussion by hipstersmoothie** [devtools-fm/devtools.fm#209] [Announcements]
- https://million.dev
- https://twitter.com/aidenybai

## Questions

- [x] Where did you come up with the idea for it? When did you develop it?
- [x] What is million.js?
- [x] Why does block level dom go so fast? What does it solve? What exactly is the optimization?
- [x] Is this a replacement for virtualization?
- [x] What does using million.js look like?
- [x] Why do I need to use the For component?
- [x] macro seems like an odd inclusion in the API. What does it do and how would I use it?
- [x] What been the toughest part of building it?
- [x] What situations does the model excel in? Where does it still struggle?
- [x] Do you see this as being a silver bullet? Or should it be selectively applied?

#### Future

- [x] How do you plan to support the project as you go into college?
- [x] What's next for the project?
- [x] What's you're spiciest dev take?
- [x] Where do you think the most exciting things in front end JS are going on?

**Comment by hipstersmoothie** [comment]
https://github.com/moonrepo/espresso
https://elysiajs.com/

**Comment by just-be-dev** [comment]
- https://github.com/gw31415/obsidian-lsp
- https://github.com/isadorasophia/murder

**Comment by aidenybai** [comment]
https://github.com/WICG/webcomponents/blob/gh-pages/proposals/DOM-Parts.md
https://github.com/naruaway/quickbootjs
